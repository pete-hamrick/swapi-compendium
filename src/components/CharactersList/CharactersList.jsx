import React from 'react'
import Characters from '../Characters/Characters'
import './CharactersList.css'

export default function CharactersList({ characters }) {
  return (
    <ul className="character-container">
      {characters.map((character) => (
        <Characters key={character.name} character={character} />
      ))}
    </ul>
  )
}
