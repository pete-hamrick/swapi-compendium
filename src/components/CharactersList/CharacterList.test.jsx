import React from 'react'
import { render } from '@testing-library/react'
import CharactersList from './CharactersList'

it('should display the character list', () => {
  const { container } = render(
    <CharactersList characters={['Dorothy', 'Rose', 'Blanche', 'Estelle']} />
  )

  expect(container).toMatchSnapshot()
})
