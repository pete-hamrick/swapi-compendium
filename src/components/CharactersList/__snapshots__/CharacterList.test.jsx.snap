// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the character list 1`] = `
<div>
  <ul
    class="character-container"
  >
    <li
      class="character"
    >
      <h5 />
      <p>
        Eye Color: 
      </p>
      <p>
        Height: 
        cm
      </p>
      <p>
        Birth Year: 
      </p>
    </li>
    <li
      class="character"
    >
      <h5 />
      <p>
        Eye Color: 
      </p>
      <p>
        Height: 
        cm
      </p>
      <p>
        Birth Year: 
      </p>
    </li>
    <li
      class="character"
    >
      <h5 />
      <p>
        Eye Color: 
      </p>
      <p>
        Height: 
        cm
      </p>
      <p>
        Birth Year: 
      </p>
    </li>
    <li
      class="character"
    >
      <h5 />
      <p>
        Eye Color: 
      </p>
      <p>
        Height: 
        cm
      </p>
      <p>
        Birth Year: 
      </p>
    </li>
  </ul>
</div>
`;
